unitTests/ui/MissingTermsTest.ts

```diff
@@ -0,0 +1,146 @@
+import * as Mock from '../MockEnvironment';
+import { MissingTerms } from '../../src/ui/MissingTerm/MissingTerms';
+import { $$ } from '../../src/Core';
+import { FakeResults } from '../Fake';
+
+export function MissingTermsTest() {
+  describe('MissingTerm', () => {
+    let test: Mock.IBasicComponentSetup<MissingTerms>;
+    let fakeResult = FakeResults.createFakeResult();
+    fakeResult.state.q = 'This will be a really really really long query';
+    const wordNotInQuery = 'foo';
+    const wordPresent = 'will';
+    const firstWord = 'This';
+    const getSpy: any = jasmine.createSpy('getSpy').and.returnValue('This will be a really really really long query');
+    beforeEach(() => {
+      fakeResult.absentTerms = [];
```

Comment:
You should modify the `FakeResults.createFakeResult()` function to always create the result absent terms with an empty array, and remove this line here.